File: backend/app/models/task.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class CreateTaskRequest(BaseModel):
    title: str
    description: Optional[str] = None
    due_date: Optional[datetime] = None
    assigned_to: Optional[str] = None
    points: int = 0


class UpdateTaskRequest(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    due_date: Optional[datetime] = None
    assigned_to: Optional[str] = None
    points: Optional[int] = None
    completed: Optional[bool] = None


class TaskResponse(BaseModel):
    id: str
    household_id: str
    title: str
    description: Optional[str] = None
    due_date: Optional[datetime] = None
    assigned_to: Optional[str] = None
    points: int
    completed: bool
    completed_at: Optional[datetime] = None
    completed_by: Optional[str] = None
    created_by: str
    created_at: datetime

File: mobile/lib/features/tasks/domain/entities/task.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'task.freezed.dart';

@freezed
class Task with _$Task {
  const factory Task({
    required String id,
    required String householdId,
    required String title,
    String? description,
    DateTime? dueDate,
    String? assignedTo,
    @Default(0) int points,
    @Default(false) bool completed,
    DateTime? completedAt,
    String? completedBy,
    required String createdBy,
    required DateTime createdAt,
  }) = _Task;
}
